### PR TITLE
Commonalize client creation to fix missing useragent

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Related Issue(s)
#870 

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/software/aws/toolkits/core/ToolkitClientManager.kt [hunk 1]
```diff
@@ -18,6 +19,7 @@ import software.amazon.awssdk.services.s3.internal.handlers.EndpointAddressInter
 import software.amazon.awssdk.services.s3.internal.handlers.PutObjectInterceptor
 import software.aws.toolkits.core.credentials.ToolkitCredentialsProvider
 import software.aws.toolkits.core.region.AwsRegion
+import sun.net.www.protocol.http.HttpURLConnection.userAgent
```
Comment (rli) [inline core/src/software/aws/toolkits/core/ToolkitClientManager.kt]: ?

File: core/src/software/aws/toolkits/core/ToolkitClientManager.kt [hunk 2]
```diff
@@ -97,37 +99,50 @@ abstract class ToolkitClientManager(private val sdkHttpClient: SdkHttpClient) {
...
+    protected open fun <T : SdkClient> createNewClient(
+        key: AwsClientKey,
+        region: AwsRegion = key.region,
+        credProvider: ToolkitCredentialsProvider = getCredentialsProvider()
+    ): T = createNewClient(key.serviceClass as KClass<T>, sdkHttpClient, Region.of(region.id), credProvider)
 
     companion object {
         private val GLOBAL_SERVICES = setOf("IamClient")
+
+        fun <T : SdkClient> createNewClient(
```
Comment (rli) [inline core/src/software/aws/toolkits/core/ToolkitClientManager.kt]: Should the telemetry client also use the new util?

https://github.com/aws/aws-toolkit-jetbrains/blob/8acb5d3f1d59df1caa2ec1223107f66d71c135ff/jetbrains-core/src/software/aws/toolkits/jetbrains/services/telemetry/TelemetryService.kt#L71

File: core/src/software/aws/toolkits/core/ToolkitClientManager.kt (hunk 1 above)
Comment (abrooksv) [reply core/src/software/aws/toolkits/core/ToolkitClientManager.kt]: Oh ya, IntelliJ auto-import burned me.

File: core/src/software/aws/toolkits/core/ToolkitClientManager.kt (hunk 2 above)
Comment (abrooksv) [reply core/src/software/aws/toolkits/core/ToolkitClientManager.kt]: I think so but looks like we will need to add endpoint to the method

**Pull request merged**